core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/content/services/html/dot-seo-meta-tags.service.ts

```diff
@@ -90,9 +104,52 @@ export class DotSeoMetaTagsService {
                     info: this.dotMessageService.get('seo.rules.title.info')
                 });
             }
+
+            if (key === SEO_OPTIONS.OG_TITLE) {
+                const items = this.getOgTitleItems(metaTagsObject);
+                const keyValues = this.getKeyValues(items);
+                result.push({
+                    key,
+                    keyIcon: keyValues.keyIcon,
+                    keyColor: keyValues.keyColor,
+                    items: items,
+                    sort: 4,
+                    info: this.dotMessageService.get('seo.rules.title.info')
+                });
+            }
+
+            if (key === SEO_OPTIONS.OG_IMAGE) {
+                const items = this.getOgImagesItems(metaTagsObject);
+                const keyValues = this.getKeyValues(items);
+                result.push({
+                    key,
+                    keyIcon: keyValues.keyIcon,
+                    keyColor: keyValues.keyColor,
+                    items: items,
+                    sort: 5,
+                    info: ''
+                });
+            }
```

Comment:
I see we are repeating this code a lot. I have this idea to just have a map object with the logic mapped by the `SEO_OPTIONS` enum. You can give it a try, it can reduce the code complexity and help us in the future if we need to add another SEO option

```javascript
enum key {
    myKey = 'myKey'
}

const keyMap = {
    //
    [key.myKey]: {
        getItems: (metaTagsObject) => getMyKeyItems(metaTagsObject),
        sort: 1,
        getInfo: () => this.dotMessageService.get('my.key.message') // I think this can be the string directly, instead of a function
    }
};

function getMyKeyItems(metaTagsObject) {
    // The get items for myKey

    return [];
}

function getKeyValues(items) {
    // Just an empty method to fullfill the example

    return {
        keyIcon: '',
        keyColor: ''
    };
}

function getKeyInfo(key, metaTagsObject) {
    // Here is the magic, this way if you need to add anothe SEO option, you just need to add a new key to the enum and add the logic in the map

    // Is just an idea, if you need help or want to discuss it, let me know

    const { getItems, sort, getInfo } = keyMap[key](metaTagsObject);

    const items = getItems(metaTagsObject);

    const { keyColor, keyIcon } = getKeyValues(items);

    return {
        items,
        sort,
        keyColor,
        keyIcon,
        info: getInfo()
    };
}

// Now you just call this function inside the Object.keys... And push the result to the array

getKeyInfo(key.myKey, { someValue: '' });

```

Reply:
Also mutate the results is not a good ide either... maybe this is just a map instead of a forEach to generate a new array.

Reply:
If this is getting bigger. I think if you need to add more Meta Tags in the near future, it's a good approach to use a map to do the work as mentioned @zJaaal .